### PR TITLE
Add createScreenshot function

### LINKED ISSUE #64: Create function for screenshots
Similar to #63 we can condense screenshots into one function

1. takeScreenshot -> returns ID
2. downloadScreenshot -> used ID

we can create `createScreenshot(TakeScreenshotInputParams)` which will take & then download the generated screenshot.

### PR DESCRIPTION
**Pull request by akash5100**
* closes #64

### CODE REVIEW

File: hvpy/helpers.py [hunk 1]
```diff
@@ -124,3 +126,83 @@ def createMovie(
...
+    x1: Optional[str] = None,
+    y1: Optional[str] = None,
+    x2: Optional[str] = None,
+    y2: Optional[str] = None,
+    watermark: Optional[bool] = False,
+    overwrite: bool = False,
+    filename: Union[str, Path] = None,
+) -> Path:
+    """
+    Automatically creates a screenshot using `takeScreenshot` functions.
```
Comment (nabobalis) [inline hvpy/helpers.py:155]:
```suggestion
    Automatically creates a screenshot using `takeScreenshot` function.
```

File: hvpy/helpers.py [hunk 2]
```diff
@@ -124,3 +126,83 @@ def createMovie(
...
+
+    Parameters
+    ----------
+    {Insert}
+    overwrite
+        Whether to overwrite the file if it already exists.
+        Default is `False`.
+    filename
+        The path to save the file to.
+        Optional, will default to ``f"{starttime}_{endtime}.{format}"``.
```
Comment (nabobalis) [inline hvpy/helpers.py:165]: I think we forgot to update this in the other PR.

File: hvpy/helpers.py [hunk 3]
```diff
@@ -124,3 +126,83 @@ def createMovie(
...
+    input_params.pop("overwrite")
+    input_params.pop("filename")
+    res = takeScreenshot(**input_params)
+    if res.get("error"):
+        raise RuntimeError(res["error"])
+    binary_data = downloadScreenshot(
+        id=res["id"],
+    )
+    if filename is None:
+        filename = f"{date.strftime('%Y_%m_%d_%H_%M_%S')}.png"
```
Comment (nabobalis) [inline hvpy/helpers.py:200]: This needs more than just the date.

File: hvpy/tests/test_helper.py [hunk 4]
```diff
@@ -93,3 +93,38 @@ def test_error_handling(tmp_path):
...
+        x0=0,
+        y0=0,
+        width=4000,
+        height=4000,
+        filename=f1,
+    )
+    assert isinstance(result, Path)
+    assert result.exists()
+    assert result == tmp_path / "screenshot.png"
+    result.unlink()  # clean up
```
Comment (nabobalis) [inline hvpy/tests/test_helper.py:114]: We don't need to do this.

File: hvpy/helpers.py (hunk 3 above)
Comment (dgarciabriseno) [reply hvpy/helpers.py:200]: agreed

Comment (nabobalis) [reply hvpy/helpers.py:200]: ID as before for now?

File: hvpy/helpers.py [hunk 5]
```diff
@@ -124,3 +126,83 @@ def createMovie(
...
+    else:
+        filename = f"{filename}.png"
+    save_file(
+        data=binary_data,
+        filename=filename,
+        overwrite=True,
```
Comment (dgarciabriseno) [inline hvpy/helpers.py:206]: This overwrite should be the overwrite from the input parameters, not hard coded to True.

File: hvpy/helpers.py (hunk 1 above)
Comment (akash5100) [reply hvpy/helpers.py:155]: Oh, I missed writing downloadScreenshot in docstring. But we can get binary data with display=True in the takeScreenshot, with this we dont have to use downloadScreenshot. Should we use only takeScreenshot or use downloadScreenshot?

Comment (nabobalis) [reply hvpy/helpers.py:155]: What's the difference?

Comment (akash5100) [reply hvpy/helpers.py:155]: No difference.

Comment (nabobalis) [reply hvpy/helpers.py:155]: Use download

File: hvpy/helpers.py [hunk 6]
```diff
@@ -124,3 +126,84 @@ def createMovie(
...
+    overwrite: bool = False,
+    filename: Union[str, Path] = None,
+) -> Path:
+    """
+    Automatically creates a screenshot using `takeScreenshot`,
+    `downloadScreenshot` functions.
+
+    Parameters
+    ----------
+    {Insert}
```
Comment (nabobalis) [inline hvpy/helpers.py:160]:
```suggestion
```

File: hvpy/helpers.py [hunk 7]
```diff
@@ -124,3 +126,84 @@ def createMovie(
...
+    overwrite
+        Whether to overwrite the file if it already exists.
+        Default is `False`.
+    filename
+        The path to save the file to.
+        Optional, will default to ``f"{res['id']}_{date.date()}.png"``.
+
```
Comment (nabobalis) [inline hvpy/helpers.py:167]:
```suggestion
        Optional, will default to ``f"{res['id']}_{date.date()}.png"``.
    {Insert}

```

File: hvpy/tests/test_helper.py [hunk 8]
```diff
@@ -93,3 +93,37 @@ def test_error_handling(tmp_path):
...
+        events=create_events(["AR"]),
+        imageScale=2.5,
+        x0=0,
+        y0=0,
+        width=4000,
+        height=4000,
+    )
+    assert isinstance(result, Path)
+    assert result.exists()
+    result.unlink()  # clean up
```
Comment (akash5100) [inline hvpy/tests/test_helper.py:129]:
```suggestion
    assert isinstance(result, Path)
    assert result.exists()
```

File: hvpy/helpers.py [hunk 9]
```diff
@@ -124,3 +126,84 @@ def createMovie(
...
     return Path(filename)
+
+
+@_add_shared_docstring(takeScreenshotInputParameters)
+def createScreenshot(
+    date: datetime,
+    imageScale: float,
+    layers: str,
+    events: Optional[str] = None,
+    eventLabels: Optional[bool] = False,
```
Comment (nabobalis) [inline hvpy/helpers.py:137]:
```suggestion
    eventLabels: bool = False,
```

File: hvpy/helpers.py [hunk 10]
```diff
@@ -124,3 +126,84 @@ def createMovie(
...
+    scale: Optional[bool] = False,
```
Comment (nabobalis) [inline hvpy/helpers.py:138]:
```suggestion
    scale: bool = False,
```

File: hvpy/helpers.py [hunk 11]
```diff
@@ -124,3 +126,84 @@ def createMovie(
...
+    height: Optional[str] = None,
+    x0: Optional[str] = None,
+    y0: Optional[str] = None,
+    x1: Optional[str] = None,
+    y1: Optional[str] = None,
+    x2: Optional[str] = None,
+    y2: Optional[str] = None,
+    watermark: Optional[bool] = False,
+    overwrite: bool = False,
+    filename: Union[str, Path] = None,
```
Comment (nabobalis) [inline hvpy/helpers.py:152]:
```suggestion
    filename: Optional[Union[str, Path]] = None,
```

File: hvpy/tests/test_helper.py [hunk 12]
```diff
@@ -93,3 +93,36 @@ def test_error_handling(tmp_path):
...
+        layers=create_layers([(DataSource.AIA_171, 100)]),
+        events=create_events(["AR"]),
+        imageScale=2.5,
+        x0=0,
+        y0=0,
+        width=4000,
+        height=4000,
+    )
+    assert isinstance(result, Path)
+    assert result.exists()
```
Comment (nabobalis) [inline hvpy/tests/test_helper.py:128]: Oh my mistake, this does not use a tmp_path, the unlink needs to be added back.

**Review by nabobalis — APPROVED** [review]

**Pull request merged**